test/modules/ModuleLib/TestResults/EdgeHubRestartDirectMethodResult.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults
+{
+    using System;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    public class EdgeHubRestartDirectMethodResult : DirectMethodTestResult
+    {
+        public EdgeHubRestartDirectMethodResult(
+            string source,
+            DateTime createdAt,
+            string trackingId,
+            Guid batchId,
+            string sequenceNumber,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
+            DateTime directMethodCompletedTime,
+            HttpStatusCode directMethodCompletedStatusCode,
+            uint restartSequenceNumber)
+            : base(
+                source,
+                createdAt,
+                trackingId,
+                batchId,
+                sequenceNumber,
+                directMethodCompletedStatusCode)
+        {
+            this.EdgeHubRestartedTime = Preconditions.CheckNotNull(edgeHubRestartedTime, nameof(edgeHubRestartedTime));
+            this.EdgeHubRestartStatusCode = Preconditions.CheckNotNull(edgeHubRestartStatusCode, nameof(edgeHubRestartStatusCode));
+            this.DirectMethodCompletedTime = Preconditions.CheckNotNull(directMethodCompletedTime, nameof(directMethodCompletedTime));
+            this.DirectMethodCompletedStatusCode = Preconditions.CheckNotNull(directMethodCompletedStatusCode, nameof(directMethodCompletedStatusCode));
+            this.RestartSequenceNumber = Preconditions.CheckNotNull(restartSequenceNumber, nameof(restartSequenceNumber));
+        }
+
+        DateTime EdgeHubRestartedTime { get; set; }
+
+        public HttpStatusCode EdgeHubRestartStatusCode { get; set; }
+
+        public DateTime DirectMethodCompletedTime { get; set; }
+
+        public HttpStatusCode DirectMethodCompletedStatusCode { get; set; }
+
+        public uint RestartSequenceNumber { get; set; }
+
+        public string GetDirectMethodTestResult()
```

Comment:
why do you need this method?

Reply:
In the TRC, I can simply deserializeObject the `.Result` into `EdgeHubRestartMessageResult` and use the method to verify it against `MessageTestResult`

Reply:
Verify if you can reuse the `GetFormattedResult()`